src/dawn/engine_dawn.cc

```diff
@@ -286,12 +291,48 @@ Result EngineDawn::DoClear(const ClearCommand*) {
   color_attachment.loadOp = ::dawn::LoadOp::Clear;
   color_attachment.storeOp = ::dawn::StoreOp::Store;
 
-  ::dawn::RenderPassDescriptor rpd =
-      device_->CreateRenderPassDescriptorBuilder()
-          .SetColorAttachments(1, &color_attachment)
-          .GetResult();
-  command_buffer_builder_.BeginRenderPass(rpd).EndPass();
-  return {};
+  ::dawn::RenderPassColorAttachmentDescriptor* rpca[] = {&color_attachment};
+  ::dawn::RenderPassDescriptor rpd;
+  rpd.colorAttachmentCount = 1;
+  rpd.colorAttachments = rpca;
+  rpd.depthStencilAttachment = nullptr;
+  ::dawn::RenderPassEncoder pass =
+      command_buffer_builder_.BeginRenderPass(&rpd);
+  pass.EndPass();
+
+  // Make sure we have a queue.
+  if (!queue_)
+    queue_ = device_->CreateQueue();
+
+  // Now run the commands.
+  ::dawn::Buffer& fb_buffer = render_pipeline_info_.fb_buffer;
+  auto command_buffer = command_buffer_builder_.Finish();
+
+  if (render_pipeline_info_.fb_data != nullptr) {
+    fb_buffer.Unmap();
+    render_pipeline_info_.fb_data = nullptr;
+  }
+
+  queue_.Submit(1, &command_buffer);
+
+  auto* pipeline = command->GetPipeline();
+  const std::vector<amber::Pipeline::BufferInfo>& out_color_attachment =
+      pipeline->GetColorAttachments();
+
+  // And any further commands start afresh.
+  DestroyCommandBufferBuilder();
+  MapResult map = MapBuffer(*device_, fb_buffer);
```

Comment:
We Unmap the fb_buffer on 312 and then re-map here? Can we just keep the map above around?

Reply:
Next PR will rework how mapping/unmapping occurs.  This buffer will be mapped only within the lifetime of this DoClear call, so it will be assumed to be unmapped at the beginning, then mapped temporarily, copied out of, then forcibly unmapped at the end again.

I'll make that change in the next PR.